rbd/snapshot_nautilus.go

```diff
@@ -65,6 +65,43 @@ func (image *Image) GetParentInfo(pool, name, snapname []byte) error {
 	return nil
 }
 
+// ParentInfo represents the parent image information.
+type ParentInfo struct {
```

Comment:
I think this type ought to more closely reflect what the ceph apis return.
This may be a new ImageSpec and SnapSpec types and then a ParentInfo type with fields containing those two types.

Reply:
Sure, but do you want me to return everything which these two structures contain or we return our own version with limited info (whatever we provide today)?

typedef struct {                                                     typedef struct {
  int64_t pool_id;                                                    uint64_t id;                  
  char *pool_name;                                                rbd_snap_namespace_type_t namespace_type;
  char *pool_namespace;                                      char *name;
  char *image_id;                                                   } rbd_snap_spec_t;
  char *image_name;
  bool trash;
} rbd_linked_image_spec_t;

Reply:
Matching (within reason) the fields that are found in the ceph structs is good, but not required for this PR. At least in the case of rbd_linked_image_spec_t, the type is used in more than one call. So eventually it would be good to use the same go-ceph type for the calls in librbd that make use of it.